src/ILVerify/src/ILImporter.Verify.cs

```diff
@@ -365,17 +365,29 @@ void CheckIsValidBranchTarget(BasicBlock src, BasicBlock target)
             if (src.TryIndex != target.TryIndex)
             {
                 if (src.TryIndex == null)
-                    VerificationError(VerifierError.BranchIntoTry);
+                {
+                    // Branching to first instruction of try-block is valid
+                    if (target.StartOffset != _exceptionRegions[(int)target.TryIndex].ILRegion.TryOffset)
+                        VerificationError(VerifierError.BranchIntoTry);
+                }
                 else if (target.TryIndex == null)
                     VerificationError(VerifierError.BranchOutOfTry);
                 else
                 {
-                    if (_exceptionRegions[(int)src.TryIndex].ILRegion.TryOffset < _exceptionRegions[(int)target.TryIndex].ILRegion.TryOffset)
-                        VerificationError(VerifierError.BranchIntoTry);
+                    var srcTryStart = _exceptionRegions[(int)src.TryIndex].ILRegion.TryOffset;
+                    var targetTryStart = _exceptionRegions[(int)target.TryIndex].ILRegion.TryOffset;
```

Comment:
This won't catch branches through multiple levels of try nesting.

This should be invalid:

```
.method private hidebysig static void foo() cil managed
{
  .try
  {
    nop
    br Inner

    .try
    {
      nop

      .try
      {
        Inner:
        leave.s    A
      }
      catch [mscorlib]System.Object 
      {
        pop
        leave.s    A
      }
      A:

      leave.s    B
    }
    catch [mscorlib]System.Object 
    {
      pop
      leave.s    B
    }
    B:

    leave.s    C
  }
  catch [mscorlib]System.Object 
  {
    pop
    leave.s    C
  }
  C:

  ret
}
```